verilog/CST/data_test.cc

```diff
@@ -0,0 +1,140 @@
+  EXPECT_OK(analyzer.Analyze());
+  const auto& root = analyzer.Data().SyntaxTree();
+  const auto data_declarations =
+      GetIdentifiersFromDataDeclaration(*ABSL_DIE_IF_NULL(root));
+  EXPECT_EQ(data_declarations.size(), 1);
+}
+
+TEST(GetIdentifiersFromDataDeclarationTest, MultipleVariables) {
+  VerilogAnalyzer analyzer("module foo; logic x; logic y; endmodule", "");
+  EXPECT_OK(analyzer.Analyze());
+  const auto& root = analyzer.Data().SyntaxTree();
+  const auto data_declarations =
+      GetIdentifiersFromDataDeclaration(*ABSL_DIE_IF_NULL(root));
+  EXPECT_EQ(data_declarations.size(), 2);
+}
+
+TEST(GetIdentifiersFromDataDeclarationTest, MultipleInlineVariables) {
+  VerilogAnalyzer analyzer("module foo; logic x, y, z; endmodule", "");
+  EXPECT_OK(analyzer.Analyze());
+  const auto& root = analyzer.Data().SyntaxTree();
+  const auto data_declarations =
+      GetIdentifiersFromDataDeclaration(*ABSL_DIE_IF_NULL(root));
+  EXPECT_EQ(data_declarations.size(), 3);
+}
+
+TEST(GetIdentifiersFromDataDeclarationTest, MultipleMixedVariables) {
+  VerilogAnalyzer analyzer("module foo; logic x, y, z; logic a; endmodule", "");
+  EXPECT_OK(analyzer.Analyze());
+  const auto& root = analyzer.Data().SyntaxTree();
+  const auto data_declarations =
+      GetIdentifiersFromDataDeclaration(*ABSL_DIE_IF_NULL(root));
+  EXPECT_EQ(data_declarations.size(), 4);
+}
+
+TEST(GetIdentifiersFromDataDeclarationTest, OneObjectVariable) {
+  VerilogAnalyzer analyzer("module top; foo baz(0); endmodule", "");
+  EXPECT_OK(analyzer.Analyze());
+  const auto& root = analyzer.Data().SyntaxTree();
+  const auto data_declarations =
+      GetIdentifiersFromDataDeclaration(*ABSL_DIE_IF_NULL(root));
+  EXPECT_EQ(data_declarations.size(), 1);
+}
+
+TEST(GetIdentifiersFromDataDeclarationTest, MultipleObjectVariables) {
+  VerilogAnalyzer analyzer("module top; foo baz(0); foo bay(1); endmodule", "");
+  EXPECT_OK(analyzer.Analyze());
+  const auto& root = analyzer.Data().SyntaxTree();
+  const auto data_declarations =
+      GetIdentifiersFromDataDeclaration(*ABSL_DIE_IF_NULL(root));
+  EXPECT_EQ(data_declarations.size(), 2);
+}
+
+TEST(GetIdentifiersFromDataDeclarationTest, MultipleInlineObjectVariables) {
+  VerilogAnalyzer analyzer("module top; foo baz(0), bay(1); endmodule", "");
+  EXPECT_OK(analyzer.Analyze());
+  const auto& root = analyzer.Data().SyntaxTree();
+  const auto data_declarations =
+      GetIdentifiersFromDataDeclaration(*ABSL_DIE_IF_NULL(root));
+  EXPECT_EQ(data_declarations.size(), 2);
+}
+
+TEST(GetIdentifiersFromDataDeclarationTest, CompleteMixOfVariables) {
+  VerilogAnalyzer analyzer(R"(
+module foo;
+foo bax(0);
+foo baz(0), bay(1);
+logic a;
+logic b, c;
+endmodule)",
+                           "");
+  EXPECT_OK(analyzer.Analyze());
+  const auto& root = analyzer.Data().SyntaxTree();
+  const auto data_declarations =
+      GetIdentifiersFromDataDeclaration(*ABSL_DIE_IF_NULL(root));
+  EXPECT_EQ(data_declarations.size(), 6);
+}
+
```

Comment:
More suggested test cases:

* array declarations: make sure `logic v[M:N]` finds only `v` and not `M` or `N`.
* declaration assignments: make sure anything with a trailing assignment doesn't match any expression identifiers from the right-hand-side expressions.

Reply:
That makes sense! Added those above.